File: wtplayer/src/WTPortal.cpp
#include "basetypes.hpp"
#include "WTPortal.hpp"
#include "WTBitmap.hpp"
#include "WTCamera.hpp"
#include "WTSurfaceShader.hpp"

WTPortal::WTPortal(WT* wt_,
                   float Width,
                   float Height,
                   WTCamera* pCam,
                   int mappingOption,
                   bool doubleSided,
                   bool renderOnlyChildren) :
  WTGroup(wt_), camera(pCam) {
  this->camera->AddRef();
}

WTPortal::~WTPortal() {
  if(this->camera)
    this->camera->Release();
  this->camera = NULL;

  if(this->texture)
    this->texture->Release();
  this->texture = NULL;

  if(this->shader)
    this->shader->Release();
  this->shader = NULL;
}

void WTPortal::setMappingOption(int mappingOption,
                                float zScale){
  APILOG;
}

void WTPortal::setCamera(WTCamera* newPCam){
  APILOG;
  if(this->camera)
    this->camera->Release();
  this->camera = newPCam;
  this->camera->AddRef();
}

void WTPortal::setDoubleSided(bool doubleSided){
  APILOG;
}

void WTPortal::setCustomTransform(int transformOption,
                                  float xAxis,
                                  float yAxis,
                                  float zAxis) {
  APILOG;
}

void WTPortal::setTexture(WTBitmap* pBitmap,
                          int newAlpha){
  APILOG;
  if(this->texture)
    this->texture->Release();
  this->texture = pBitmap;
  this->texture->AddRef();
}

void WTPortal::setRenderOnlyChildren(bool renderOnlyChildren){
  APILOG;
}

void WTPortal::setSurfaceShader(WTSurfaceShader* pShader,
                                int newAlpha){
  APILOG;
  if(this->shader)
    this->shader->Release();
  this->shader = pShader;
  this->shader->AddRef();
}

File: wtplayer/javasrc/wildtangent/webdriver/WTPortal.java
package wildtangent.webdriver;

import wildtangent.webdriver.WTBitmap;
import wildtangent.webdriver.WTCamera;
import wildtangent.webdriver.WTGroup;
import wildtangent.webdriver.WTSurfaceShader;

public interface WTPortal
extends WTGroup {
    public void setMappingOption(int var1);

    public void setTexture(WTBitmap var1);

    public void setCustomTransform(int var1, float var2, float var3, float var4);

    public void setDoubleSided(boolean var1);

    public void setSurfaceShader(WTSurfaceShader var1, int var2);

    public void setSurfaceShader(WTSurfaceShader var1);

    public void setRenderOnlyChildren(boolean var1);

    public void setCamera(WTCamera var1);

    public WTCamera getCamera();

    public void setTexture(WTBitmap var1, int var2);

    public void setMappingOption(int var1, float var2);
}

